### ISSUE TITLE
Unable to snatch when "only download "FreeLeech" torrents." is set

### ISSUE THREAD

**Issue by GritGub** [pymedusa/Medusa#5344]
When set "only download "FreeLeech" torrents." in menu 'Search Providers' Medusa is unable to find _any_ torrents. 
I can confirm positive that a given episode is available for FreeLeech, but Medusa results negative. As soon as I disable the flag "only download "FreeLeech" torrents.", medusa manual search an auto backlog search works ok.

The log do not register the error.
Very easy to reproduce this error - If any screenshots is needed, please let me know.

Private Tracker: DanishBits.org

Medusa info:

Medusa Info: | Branch: master 
Commit: 4614efc77151ded92ef458a09dec39f8bd5acfc6 
Version: v0.2.10 
Database: 44.11
Python Version: | 2.7.14 (default, May 28 2018, 14:47:18) [GCC 4.7.3 20130102 (prerelease)]
SSL Version: | OpenSSL 1.0.2n 7 Dec 2017
OS: | Linux-3.10.105-x86_64-with-glibc2.2.5
Locale: | Unknown.Unknown

**Comment by medariox** [comment]
Maybe they changed something in their API? Please post the result of an API search with a freeleech torrent.

**Comment by GritGub** [comment]
I am not sure what you mean by API search... I have tried to do several Manual search etc. but same result... I have also tried on two different private trackers, independent of each other. Same result.

**Comment by GritGub** [comment]
Problem still excist... Medusa is able to find all 'Wanted' episodes when 'Only download FreeLeech' is **disabled**, as soon as this setting is set to enable in Search Provider setting, then Medusa finds zero results. I have talked to the danishbits devs, and thy claim that this is a Medusa issue.

LOG for Forced search - If I do the same when FreeLeech is disabled, then Medusa finds several episodes, also eposide marked with freeleech on danishbits:

2019-01-12 13:48:56 INFO     SEARCHQUEUE-FORCED-260449 :: [918cfe7] Unable to find forced results for: Vikings - S05E17 - The Most Terrible Thing
2019-01-12 13:48:56 INFO     SEARCHQUEUE-FORCED-260449 :: [Danishbits] :: [918cfe7] Performing season pack search for Vikings
2019-01-12 13:48:29 INFO     SEARCHQUEUE-FORCED-260449 :: [Danishbits] :: [918cfe7] Performing episode search for Vikings
2019-01-12 13:48:29 INFO     SEARCHQUEUE-FORCED-260449 :: [918cfe7] Using backlog search providers
2019-01-12 13:48:29 INFO     SEARCHQUEUE-FORCED-260449 :: [918cfe7] Building internal name cache for Vikings
2019-01-12 13:48:29 INFO     SEARCHQUEUE-FORCED-260449 :: [918cfe7] Finished processing 3014 scene exceptions.
2019-01-12 13:48:29 INFO     SEARCHQUEUE-FORCED-260449 :: [918cfe7] Updating exception_cache and exception_season_cache
2019-01-12 13:48:29 INFO     SEARCHQUEUE-FORCED-260449 :: [918cfe7] Beginning forced search for: Vikings - S05E17 - The Most Terrible Thing

**Comment by medariox** [comment]
Let me try again. We have no access to the provider. This means you need to query their API and post a result that contains at last one freeleech item.

**Comment by p0psicles** [comment]
Closed until requested info is available

**Comment by GritGub** [comment]
As I stated to you last time, I have no idea what you are talking about - I am unable to query their API as I do not know what you talk about. What I _can_ do is post you an invite to danishbits and you are then free to test it yourself. If not, then please post a guide on how to perform this query API, and I will test it as good as I can.

**Comment by medariox** [comment]
@GritGub 
In your browser you need to enter:
```
https://danishbits.org/couchpotato.php?user=YOUR_USERNAME&passkey=YOUR_PASSKEY&search=YOUR_FREELEECH_SHOW&latest=false
```
YOUR_FREELEECH_SHOW = Show+Name+S05E05

Post the result and MAKE SURE at least one freeleech item is returned.

**Comment by GritGub** [comment]
@medariox Thank you for a good guide.

This is the result when I replace YOUR_FREELEECH_SHOW with "Vikings.S05E17.NORDiC.1080p.WEB-DL.DDP5.1.H.264-DBRETAiL" that I know is a freeleech episode.
Please note that the output result from Danishbits shows that this episode is not a freeleech - I guess that is why Medusa return 0.... I have also attached screenshot from the filename.

![image](https://user-images.githubusercontent.com/25592744/51116967-c7118880-180c-11e9-8e18-1e98effbb341.png)


:::::::::::::::::::::::::::::
{"results":[{"release_name":"Vikings.S05E17.NORDiC.1080p.WEB-DL.DDP5.1.H.264-DBRETAiL","torrent_id":"1167994","details_url":"https:\/\/danishbits.org\/torrents.php?id=1169052","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.NORDiC.1080p.WEB-DL.DDP5.1.H.264-DBRETAiL.torrent?action=download&authkey=XXXREMOVEDBYME&torrent_pass=XXXREMOVEDBYME&id=1167994","imdb_id":"tt2306299","freeleech":false,"type":"tv","size":1526,"leechers":16,"seeders":2650,"publish_date":"2019-01-10 12:39:51"}],"total_results":1}
::::::::::::::::::::::::::::

And this is the result when I replace the YOUR_FREELEECH_SHOW with "Vikings.S05E17"

::::::::::::::::::::::::::
{"results":[{"release_name":"Vikings.S05E17.720p.HDTV.x264-AVS","torrent_id":"1167558","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168616","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.720p.HDTV.x264-AVS.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167558","imdb_id":"0","freeleech":false,"type":"tv","size":781,"leechers":11,"seeders":95,"publish_date":"2019-01-10 04:03:44"},{"release_name":"Vikings.S05E17.HDTV.x264-SVA","torrent_id":"1167560","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168618","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.HDTV.x264-SVA.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167560","imdb_id":"0","freeleech":false,"type":"tv","size":244,"leechers":8,"seeders":15,"publish_date":"2019-01-10 04:04:10"},{"release_name":"Vikings.S05E17.720p.HEVC.x265-MeGusta","torrent_id":"1167572","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168630","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.720p.HEVC.x265-MeGusta.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167572","imdb_id":"0","freeleech":false,"type":"tv","size":184,"leechers":0,"seeders":27,"publish_date":"2019-01-10 04:36:49"},{"release_name":"Vikings.S05E17.READNFO.WEB.h264-TBS","torrent_id":"1167836","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168894","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.READNFO.WEB.h264-TBS.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167836","imdb_id":"0","freeleech":false,"type":"tv","size":624,"leechers":1,"seeders":2,"publish_date":"2019-01-10 09:29:14"},{"release_name":"Vikings.S05E17.READNFO.720p.WEB.h264-TBS","torrent_id":"1167838","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168896","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.READNFO.720p.WEB.h264-TBS.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167838","imdb_id":"0","freeleech":false,"type":"tv","size":849,"leechers":0,"seeders":15,"publish_date":"2019-01-10 09:33:23"},{"release_name":"Vikings.S05E17.1080p.WEB.H264-METCON","torrent_id":"1167848","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168906","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.1080p.WEB.H264-METCON.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167848","imdb_id":"0","freeleech":false,"type":"tv","size":1546,"leechers":0,"seeders":50,"publish_date":"2019-01-10 09:44:04"},{"release_name":"Vikings.S05E17.iNTERNAL.720p.HEVC.x265-MeGusta","torrent_id":"1167849","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168907","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.iNTERNAL.720p.HEVC.x265-MeGusta.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167849","imdb_id":"0","freeleech":false,"type":"tv","size":196,"leechers":-3,"seeders":17,"publish_date":"2019-01-10 09:47:15"},{"release_name":"Vikings.S05E17.iNTERNAL.720p.WEB.H264-AMRAP","torrent_id":"1167854","details_url":"https:\/\/danishbits.org\/torrents.php?id=1168912","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.iNTERNAL.720p.WEB.H264-AMRAP.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167854","imdb_id":"0","freeleech":false,"type":"tv","size":689,"leechers":1,"seeders":34,"publish_date":"2019-01-10 09:52:26"},{"release_name":"Vikings.S05E17.NORDiC.1080p.WEB-DL.DDP5.1.H.264-DBRETAiL","torrent_id":"1167994","details_url":"https:\/\/danishbits.org\/torrents.php?id=1169052","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.NORDiC.1080p.WEB-DL.DDP5.1.H.264-DBRETAiL.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167994","imdb_id":"tt2306299","freeleech":false,"type":"tv","size":1526,"leechers":15,"seeders":2649,"publish_date":"2019-01-10 12:39:51"},{"release_name":"Vikings.S05E17.NORDiC.720p.WEB-DL.DDP5.1.H.264-DBRETAiL","torrent_id":"1167995","details_url":"https:\/\/danishbits.org\/torrents.php?id=1169053","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.NORDiC.720p.WEB-DL.DDP5.1.H.264-DBRETAiL.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1167995","imdb_id":"tt2306299","freeleech":false,"type":"tv","size":679,"leechers":1,"seeders":840,"publish_date":"2019-01-10 12:40:03"},{"release_name":"Vikings.S05E17.PROPER.HDTV.x264-W4F","torrent_id":"1168158","details_url":"https:\/\/danishbits.org\/torrents.php?id=1169216","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.PROPER.HDTV.x264-W4F.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1168158","imdb_id":"0","freeleech":false,"type":"tv","size":251,"leechers":1,"seeders":5,"publish_date":"2019-01-10 19:38:34"},{"release_name":"Vikings.S05E17.PROPER.720p.HDTV.x264-W4F","torrent_id":"1168159","details_url":"https:\/\/danishbits.org\/torrents.php?id=1169217","download_url":"https:\/\/danishbits.org\/torrents.php\/Vikings.S05E17.PROPER.720p.HDTV.x264-W4F.torrent?action=download&authkey=REPLACEDBYME&torrent_pass=REPLACEDBYME&id=1168159","imdb_id":"0","freeleech":false,"type":"tv","size":907,"leechers":-1,"seeders":36,"publish_date":"2019-01-10 19:39:05"}],"total_results":12}
::::::::::::::::::::::::::::::::::::

**Comment by medariox** [comment]
There we go. As you have correctly identified yourself the issue is on Danishbits' side. If they don't report the correct freeleech as true we obviously can't know it's a freeleech torrent.

**Comment by GritGub** [comment]
After I spoke to staff on danishbits.org I can confirm that the API search string on the site is indeed "broken", all of their search result returns: "freeleech":false - as they use their own built-in freeleech category system and do not set this in the API. 

There are currently no fix for this issue.
